### DISCUSSION TITLE
Alpine JS v3 iOS 10.3.3

### DISCUSSION THREAD

**Discussion by makingthingswork** [alpinejs/alpine#2420] [7. Bugs (archived)]
In case anyone else encounters this I thought I'd post this as it took me a little time to figure out exactly what was happening.

I've noticed that v3 doesn't work at all on iOS 10.3.3. I happened to be using a 4th generation iPad and decided to view some sites I had built in recent times. Admittedly this is an old piece of hardware but it had the most up to date iOS (10.3.3) possible on that machine as it no longer updates. Nothing JS related worked at on all my Alpine JS sites (2 productions builds and one site soon to launch) as all had v3. The two production sites were both built earlier this year in v2 before v3 launched but were upgraded with no  known issue at the time.

The two production sites I worked on finding the issue but the site in development is quite simple so I decided to swap out the Alpine JS for vanilla JS so I just recoded the JS parts on a new branch if it is needed.

Rolling back to v2 and replacing any use of `x-data={ ...function_name_here() }` with `x-data="function_name_here()"` corrected all the Alpine JS problems. I wasn't using multiple data objects anywhere so the latter wasn't needed. In addition, I wasn't using anything that was specific to v3 so it all works again.

Is this a bug?! Not sure but the only thing I could find anywhere in the docs relating to browser support was in relation to IE11. It would definitely be useful to have a reference with regards to support/non-support. This specific issue would probably impact on a tiny fraction of users but all the navigation for iOS devices was dependent on JS so it makes the sites unusable.

**Comment by danddanddand** [comment]
My site for sure works on iOS with v3 so it's probably something with your build or code. Can you see the error that is happening? What's the url?

**Reply by danddanddand** [reply]
Actually iOS 10 maybe isn't supported I don't have old enough device to test.

**Reply by makingthingswork** [reply]
What version of iOS? I had no issue at all with recent versions but that specific older version mentioned in the original post meant the site went from working as expected to not working at all. Its not really a problem as I have no requirement to use v3.

**Comment by SimoTod** [comment]
Alpine v3 relies on queueMicrotask so you need a polyfill for iOS 10
https://developer.mozilla.org/en-US/docs/Web/API/queueMicrotask#when_queuemicrotask_isnt_available

To answer your next question, yeah, the docs should mention it. Feel free to send a PR.

**Reply by makingthingswork** [reply]
I'll look into that, thanks for the advice. 

Interesting that `queueMicrotask` is only supported in Safari iOS 12.2 or newer and 12.2 was released in March 2019 so it's relatively new. v3 was released in June so that browser was only just a little over two years old at that point.

**Reply by smappio** [reply]
Hello,

I'm facing a very strange issue with iphone/safari. Using alpine v3, only part of some alpine data seem to be rendered, some others are even bit defined. Using chrome or Edge, all is OK.

I also tested using Lambda Test, and same result : for non Ios devices it works, else thé problem is there.

Is there something "obvious" that could explain this problem with alpine v3 vs IOS devices & safari ?

Many thanks for your help, I'm a bit disappointed as the launch date is soon but without my web app working on ios, I won't be able to launch..

Chris
